### PR TITLE
nf: further refactoring ModelsV16 to match ModelManager public API surface 

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
See #9465 #8988

Refactorings and bugfixes, not done yet (one more PR), but there's quite a lot of code to get through, and breaking it up helps

## How Has This Been Tested?
Nonfunctional refactorings only - should be unit testable

## Learning 
Kotlin does not have `throws` (it has an annotation for Java, but checked exceptions don't exist)

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Storage.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ private static void _upgrade(Collection col, int ver) {
             if (ver < 4) {
                 col.modSchemaNoCheck();
                 List<Model> models = col.getModels().all();
-                ArrayList<Model> clozes = new ArrayList<>(models);
+                ArrayList<Model> clozes = new ArrayList<>(models.size());
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Storage.java:165]: nice catch!

File: AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt [hunk 2]
```diff
@@ -95,7 +95,7 @@ var NoteType.sortf: int
 var NoteType.type: Int
     get() = getInt("type")
     set(value) {
-        put("typr", value)
+        put("type", value)
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt:98]: ugh - is there anyway to make this a constant?

**Review by mikehardy — APPROVED** [review]
one question about perhaps making "type" a constant to perma-fix the typo-bug fixed, but that's all PR scope creep, feel free to peel to separate issue and merge, all the commits are fine by my read

File: AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt (hunk 2 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt:98]: After we're done, the only place it'll be used is here. I'm not sure if it'll be worth the extraction

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt:98]: `#insert <Mandalorian/IHaveSpoken.h>`

**Pull request merged**